### PR TITLE
chore: upgrade crossbeam channel

### PR DESCRIPTION
**Pull request by driftluo**
release note：
   -  Fix a bug in reference counting.
   - Optimize recv_timeout().
   - Add Select::remove().
   - Various small improvements, code cleanup, more tests.

### CODE REVIEW

File: chain/src/chain.rs [hunk 1]
```diff
@@ -176,6 +176,7 @@ impl<CS: ChainStore + 'static> ChainService<CS> {
         ChainService { shared, notify }
     }
 
+    #[allow(clippy::zero_ptr, clippy::drop_copy)]
```
Comment (doitian) [inline chain/src/chain.rs]: Please add a comment why exclude these linter rules.

Comment (doitian) [reply chain/src/chain.rs]: I think we should add a wrapper method to create the crossbeam channel temporarily, and add flags to that wrapper method.

**Comment by quake** [comment]
reference issue: https://github.com/crossbeam-rs/crossbeam/issues/404

File: chain/src/chain.rs (hunk 1 above)
Comment (driftluo) [reply chain/src/chain.rs]: > I think we should add a wrapper method to create the crossbeam channel temporarily, and add flags to that wrapper method.

I think this behavior has no sense other than adding complexity. `select!` is a Macro with uncertain parameters, which means that if you want to wrap it, you also need a Macro with uncertain parameters, macro set macros, adding complexity, and almost no income

**Pull request merged**